NotesApp/NotesApp/Model/NotesService.swift

```diff
@@ -8,15 +8,134 @@
 import Foundation
 
 class NotesService {
+    private let fileManager: FileManager
+    private let path: URL
+    private let decoder = JSONDecoder()
+    private let encoder = JSONEncoder()
+    private let userDefaults = UserDefaults.standard
+    private var pinnedNotesKey = "pinnedNotesKey"
+    
+    init() {
+        fileManager = FileManager()
+        path = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first!
+    }
+    
     /// Получение списка всех заметок.
     func notes(completionHandler: @escaping (Result<[Note], Error>) -> Void) {
+        var notes: [Note] = []
+        var pinnedNotes: [Note] = []
+        let pinnedNotesId = userDefaults.stringArray(forKey: pinnedNotesKey) ?? [String]()
+        do {
+            for content in try fileManager.contentsOfDirectory(atPath: path.path) {
+                let filePath = path.appendingPathComponent(content)
+                guard let data = fileManager.contents(atPath: filePath.path) else { return }
+                let note = decodeNote(data: data)
+                if pinnedNotesId.contains(note.id.uuidString) {
+                    pinnedNotes.append(note)
+                } else {
+                    notes.append(note)
+                }
+            }
+        } catch {
+            completionHandler(.failure(error))
+        }
+        pinnedNotes = pinnedNotes.sorted(by: { n1, n2 in
+            return n1.dateModified! > n2.dateModified!
+        })
+        notes = notes.sorted(by: { n1, n2 in
+            return n1.dateModified! > n2.dateModified!
+        })
```

Comment:
Можно немного сократить
```suggestion
        pinnedNotes = pinnedNotes.sorted { n1, n2 in
            return n1.dateModified! > n2.dateModified!
        }
        notes = notes.sorted { n1, n2 in
            return n1.dateModified! > n2.dateModified!
        }
```